libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -202,11 +202,11 @@ string IRGenerator::generateGetter(VariableDeclaration const& _varDecl)
 
 	Type const* type = _varDecl.annotation().type;
 
-	solAssert(!_varDecl.isConstant(), "");
 	solAssert(_varDecl.isStateVariable(), "");
 
 	if (auto const* mappingType = dynamic_cast<MappingType const*>(type))
 		return m_context.functionCollector().createFunction(functionName, [&]() {
+			solAssert(!_varDecl.isConstant(), "");
```

Comment:
Are immutable mappings supported? If not maybe add another assert here for !immutable.

Reply:
not sure what an immutable mapping should be, but I can add the assertion.

Reply:
https://github.com/ethereum/solidity/pull/8871

Reply:
Yeah... I thought about pointing towards this when approving, but then wasn't sure there can even be immutable mappings, so I ignored it :-).

Reply:
> not sure what an immutable mapping should be,

The same as other immutable value types (e.g. set in the constructor only)?